src/main/resources/test_files/runtime/Castable.jq

```diff
@@ -0,0 +1,22 @@
+(:JIQS: ShouldRun; Output="(true, true, true, true, true, true, false, true, false, true, true, false, true, false, true, true, true, true, true)" :)
+1 castable as integer,
+2.14 castable as decimal,
+1 castable as decimal,
+3.4 castable as double,
+(4 + 2.3e5) castable as decimal,
+3e4 castable as integer,
+"aa" castable as integer,
+"aa" castable as string,
+(1,2,3) castable as integer,
+false castable as boolean,
+2 castable as boolean,
+null castable as string?,
+() castable as string?,
+() castable as string,
+"false" castable as boolean,
+"3e4" castable as double,
+"3" castable as decimal,
+3 castable as string,
+3.4 castable as boolean
+
```

Comment:
and for example null castable as null (true), "null" castable as null (true), "foo" castable as null (false), 2 castable as null (false).

The null type is formally defined in the spec as a type with a singleton value space containing only the null value, associated with the literal "null". You will not find it in the W3C website because this is introduced by JSONiq.

Reply:
_Added the new tests._